test-support/src/main/java/org/springframework/ldap/test/TestContextSourceFactoryBean.java

```diff
@@ -127,6 +152,30 @@ protected Object createInstance() throws Exception {
 			LdapTestUtils.clearSubContexts(contextSource, LdapUtils.newLdapName(defaultPartitionSuffix));
 		}
 
+		if (!CollectionUtils.isEmpty(additionalSchemas)) {
+			DirContext context = contextSource.getReadWriteContext();
+			SearchControls controls = new SearchControls();
+			controls.setSearchScope(SearchControls.SUBTREE_SCOPE);
+
+			Name schemasBase = LdapUtils.newLdapName(SCHEMAS_BASE);
+			for (String schemaName : additionalSchemas) {
+				String schemaSearch = String.format(SCHEMAS_SEARCH_FORMAT, schemaName);
```

Comment:
We should not be using String format for creating attributes as values are not property escaped. We should instead use something like `LdapQueryBuilder` http://docs.spring.io/spring-ldap/docs/current/reference/#basic-queries